src/accelerate/state.py

```diff
@@ -89,6 +89,7 @@ def __init__(
             dynamo_backend = (
                 parse_choice_from_env("ACCELERATE_DYNAMO_BACKEND", "no") if dynamo_backend is None else dynamo_backend
             )
+            env_device = os.environ.get("ACCELERATE_TORCH_DEVICE", None)
```

Comment:
This can be refactored as:

```suggestion
            env_device = os.environ.get("ACCELERATE_TORCH_DEVICE", None)
            if  env_device is not None:
                self.device = torch.device(env_device)
```

And then in each `self.device` check you can reduce it to

```python
if env_device is None:
   self.device = ...
```

Reply:
Or `self.device = torch.device(env_device) if env_device is not None else None` here and then all checks are
```
if self.device is None:
    self.device = ...
```

Reply:
Agreed :) Was debating back and forth on that option, but if you were thinking that as well I'm in agreement

Reply:
Sounds good!